### PR TITLE
Improve tests

### PR DESCRIPTION
**Pull request by furstenheim-geoblink**
I've improved testing so that we can have several tests files and tests cases.

Also, bumped ts-morph version

Checked how to do array types

### CODE REVIEW

File: test/test.ts [hunk 1]
```diff
@@ -1,13 +1,34 @@
...
+  }
+]
 describe('Apidoc TS Plugin', () => {
-  it('Should include TS info in compiled api docs', () => {
-    expect(outputJson).to.deep.equal(fixtureJson)
+  tests.forEach(function (test) {
+    (test.only ? it.only : (test.skip ? it.skip : it))(test.name, async function () {
+      const dest = `out/${test.name}`
+      apidoc.createDoc({
+        src: `${test.name}`, debug: false, dest })
```
Comment (Sumolari) [inline test/test.ts]: This is weirdly indented

File: test/test.ts [hunk 2]
```diff
@@ -1,13 +1,34 @@
...
+import * as apidoc from 'apidoc'
+import * as fs from 'fs-extra'
+import * as path from 'path'
+const tests = [
+  {
+    only: false,
+    name: 'test1'
+  },
+  {
+    skip: true,
```
Comment (fofi) [inline test/test.ts]: why skipping some tests?

Comment (furstenheim-geoblink) [reply test/test.ts]: Because they are broken. But we keep them to know what does not work. (For example Array<MyClass>)

File: test/test.ts (hunk 1 above)
Comment (furstenheim-geoblink) [reply test/test.ts]: I don't really see a problem (nor the linter) but changed

**Pull request merged**